core/templates/dev/head/domain/editor/undo_redo/UndoRedoService.js

```diff
@@ -93,180 +98,73 @@ oppia.factory('UndoRedoService', [
-     * EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED.
-     */
-    QuestionUndoRedoService.undoChange = function(domainObject) {
-      if (_appliedChanges.length !== 0) {
-        var change = _appliedChanges.pop();
-        _undoneChanges.push(change);
-        _reverseChange(change, domainObject);
-        return true;
-      }
-      return false;
-    };
-
-    /**
-     * Reverses an undo for the given domain object. This function returns false
-     * if there are no changes to redo, and true if otherwise. This will fire an
-     * event as defined by the constant EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED.
-     */
-    QuestionUndoRedoService.redoChange = function(domainObject) {
-      if (_undoneChanges.length !== 0) {
-        var change = _undoneChanges.pop();
-        _appliedChanges.push(change);
-        _applyChange(change, domainObject);
-        return true;
-      }
-      return false;
-    };
-
-    /**
-     * Returns the current list of changes applied to the provided domain
-     * object. This list will not contain undone actions. Changes to the
-     * returned list will not be reflected in this service.
-     */
-    QuestionUndoRedoService.getChangeList = function() {
-      // TODO(bhenning): Consider integrating something like Immutable.js to
-      // avoid the slice here and ensure the returned object is truly an
-      // immutable copy.
-      return _appliedChanges.slice();
-    };
-
-    /**
-     * Returns a list of commit dict updates representing all chosen changes in
-     * this service. Changes to the returned list will not affect this service.
-     * Furthermore, the returned list is ready to be sent to the backend.
-     */
-    QuestionUndoRedoService.getCommittableChangeList = function() {
-      var committableChangeList = [];
-      for (var i = 0; i < _appliedChanges.length; i++) {
-        committableChangeList[i] = _appliedChanges[i].getBackendChangeObject();
-      }
-      return committableChangeList;
-    };
-
-    QuestionUndoRedoService.setChangeList = function(changeList) {
-      _appliedChanges = angular.copy(changeList);
-    };
-
-    /**
-     * Returns the number of changes that have been applied to the domain
-     * object.
-     */
-    QuestionUndoRedoService.getChangeCount = function() {
-      return _appliedChanges.length;
-    };
-
-    /**
-     * Returns whether this service has any applied changes.
-     */
-    QuestionUndoRedoService.hasChanges = function() {
-      return _appliedChanges.length !== 0;
-    };
-
-    /**
-     * Clears the change history. This does not reverse any of the changes
-     * applied from applyChange() or redoChange(). This will fire an event as
-     * defined by the constant EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED.
-     */
-    QuestionUndoRedoService.clearChanges = function() {
-      _appliedChanges = [];
-      _undoneChanges = [];
-      _dispatchMutation();
-    };
+oppia.factory('UndoRedoService', [
```

Comment:
Should this be named ExplorationUndoRedoService?

Reply:
I don't think so, it's used in the topic editor, for example.